util/types/fsp.go

```diff
@@ -75,10 +85,16 @@ func parseFrac(s string, fsp int) (int, error) {
 		round = math.Floor(digit)
 	}
 
+	if round >= math.Pow10(fsp) {
+		// overflow
+		return 0, true, nil
+	}
+
 	// Get the final frac, with 6 digit number
 	//  0.1236 round 3 -> 124 -> 123000
 	//  0.0312 round 2 -> 3 -> 30000
-	return int(round * math.Pow10(MaxFsp-fsp)), nil
+	//  0.9999999 round 6 -> 1000000 -> overflow
```

Comment:
this would not be included in this PR, it not diff master, but diff
tiancaiamao/timestamp-no-zero from tiancaiamao/time-format